src/AnnotationReader.php

```diff
@@ -279,11 +279,9 @@ static function ($attribute) {
     }
 
     /**
-     * @param ReflectionMethod|ReflectionProperty $reflection
-     *
      * @throws AnnotationException
      */
-    public function getMiddlewareAnnotations($reflection): MiddlewareAnnotations
+    public function getMiddlewareAnnotations(ReflectionMethod|ReflectionProperty $reflection): MiddlewareAnnotations
```

Comment:
We still need to support php7 where this syntax is not available. Same below, `mixed` is a new feature too. Could you revert them all, please?

Reply:
Sure. However tests did not run on php8 without this test. And running with php 7.4 said there where dependencies requiring php 8 so that didn’t work either.

Reply:
Sorry for that - I mostly run PHP 8 locally and it seems CodeSniffer figured typings would be better in the function definition.
Having reverted that change, the others also seem to have become obsolete.
I have force-pushed this branch to clean up the mess caused by that. So if you have pulled to review, please take that into account.